stan/math/rev/fun/quad_form.hpp

```diff
@@ -93,6 +94,65 @@ class quad_form_vari : public vari {
 
   quad_form_vari_alloc<Ta, Ra, Ca, Tb, Rb, Cb>* impl_;
 };
+
+/**
+ * Return the quadratic form \f$ B^T A B \f$.
+ *
+ * Symmetry of the resulting matrix is not guaranteed due to numerical
+ * precision.
+ *
+ * @tparam Mat1 type of the first (square) matrix
+ * @tparam Mat2 type of the second matrix
```

Comment:
More informative docs would be good here

Reply:
It is hard to be more specific. Both arguments are matrix types in the sense that they satisfy `require_matrix_t`, and one of them needs to be a reverse mode autodiff type.

We could say inherits from `EigenBase` or is a `var_value<T>` with `T` such that it inherits from `EigenBase` but that is sorta just redefining `require_matrix_t` in the docs.

Suggestions?

Reply:
> We could say inherits from EigenBase or is a var_value<T> with T such that it inherits from EigenBase but that is sorta just redefining require_matrix_t in the docs.

I usually write that out. If someone is one the website the requires don't show up

Reply:
More docs added